src/test/java/testresults/PostTestResultsLEC.java

```diff
@@ -427,4 +430,400 @@ public void testResultsAPIConsumerErrorLECNoFuelType() {
+        testResultsSteps.validateData("Test records created");
+
+        // Retrieve the created record, and verify that the fields are present.
+        testResultsSteps.getTestResults(randomVin);
+        testResultsSteps.statusCodeShouldBe(200);
+
+        // Verify LEC test field values match the expected values.
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "45");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC)");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "lev");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+    }
+
+    @WithTag("In_Test")
+    @Title("CVSB-7521 - TC - AC1 - Generate certificate (PASS) - LEC with linked test - HGV")
+    @Test
+    public void testResults_LEC_Linked_Test_HGV_Pass_Certificate_Number() {
+
+        // Read the base test result JSON.
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_LEC_linked_test_hgv_pass_7521.json", "$");
+
+        // Create alteration to add one more tech record to in the request body
+        String randomVin = GenericData.generateRandomVin();
+        String randomTestResultId = UUID.randomUUID().toString();
+        JsonPathAlteration alterationVin = new JsonPathAlteration("$.vin", randomVin, "", "REPLACE");
+        JsonPathAlteration alterationTestResultId = new JsonPathAlteration("$.testResultId", randomTestResultId, "", "REPLACE");
+        JsonPathAlteration alterationResult = new JsonPathAlteration("$.testTypes[0].testResult", "pass", "", "REPLACE");
+
+        // Collate the list of alterations.
+        List<JsonPathAlteration> alterations = new ArrayList<>(Arrays.asList(
+                alterationVin,
+                alterationResult,
+                alterationTestResultId));
+
+        // Post the results, together with any alterations, and verify that they are accepted.
+        testResultsSteps.postVehicleTestResultsWithAlterations(testResultRecord, alterations);
+        testResultsSteps.statusCodeShouldBe(201);
+        testResultsSteps.validateData("Test records created");
+
+        // Retrieve the created record, and verify that the fields are present.
+        testResultsSteps.getTestResults(randomVin);
+        testResultsSteps.statusCodeShouldBe(200);
+
+        // Verify LEC test field values match the expected values.
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeId", "44");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testTypeName", "Low Emissions Certificate (LEC) with annual test");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].testCode", "ldv");
+        testResultsSteps.valueForFieldInPathShouldBe("[0].testTypes[0].certificateNumber", testResultsSteps.getTestNumber());
+    }
+
+
+    @WithTag("In_Test")
```

Comment:
remove this line once all the tests for the feature are passing in CI

Reply:
Test cases executed in feature, please refer - https://jenkins.cvs.dvsacloud.uk/job/UPDATE__BRANCH/job/job_feature_test_backend/277/

In_Test tag removed.